src/main/MQ2ImGuiConsole.cpp

```diff
@@ -1554,6 +1587,39 @@ static void MakeColorGradient(float frequency1, float frequency2, float frequenc
 	WriteChatColor(szBuffer);
 }
 
+void FontSizePicker()
+{
+	for (int i = 0; i < s_fontSizeInfo.size(); ++i)
+	{
+		if (s_fontSizeInfo[i].value == s_fontSize)
+		{
+			s_currentFontIndex = i;
+			break;
+		}
+	}
+
+	if (ImGui::Combo("Main Console Font Size", &s_currentFontIndex,
+		[](void* data, int idx, const char** out_text) {
+			*out_text = static_cast<FontSizeInfo*>(data)[idx].label.c_str();
+			return true;
+		}, s_fontSizeInfo.data(), static_cast<int>(s_fontSizeInfo.size())))
+	{
+		int selectedFontSize = s_fontSizeInfo[s_currentFontIndex].value;
+
+		if (selectedFontSize != s_fontSize)
+		{
+			s_fontSize = selectedFontSize;
+			if (gImGuiConsole)
+				gImGuiConsole->m_zepEditor->SetFontSize(s_fontSize);
+
+			WritePrivateProfileInt("Console", "FontSize", s_fontSize, internal_paths::MQini);
+		}
+	}
```

Comment:
This is an overly complex way to do a combo -- it's probably better to just use the standard BeginCombo/EndCombo pattern standard to imgui. I think this is reflective of how you have set up data, which should be reconsidered

Reply:
I don't think this has anything to do with how the data is stored, but BeginCombo/EndCombo is much simpler.